### PR TITLE
Bundle Preload files into a Module

### PR DESCRIPTION
**Pull request by hellofromtonya**
The files for Preload are distributed through the plugin. This PR bundles them together by:

- Pulling all of them into one module, i.e. `inc/Engine/Preload`
- Changes the namespaces to match
- Renames the class files to PSR-4, i.e. to match the class names

## Why

This PR is part of our Dev initiative to rearchitect Rocket's file structure into modular, reusable packages and modules. It complies with our coding standards.

Benefits:

- Easier to see the relationships between the files
- Easier to work with, navigate, refactor, troubleshoot, etc. as everything is bundled together
- PSR-4 gives us a modern PHP approach to naming and fits seamlessly into computer PSR-4 autoloader

## TODO

- [x] Live test -- tested on knowthecode.dev. Works.
- [x] QA

### CODE REVIEW

File: inc/Engine/Preload/preload.php [hunk 1]
```diff
@@ -2,6 +2,10 @@
```
Comment (Tabrisrp) [inline inc/Engine/Preload/preload.php]: This file should be in inc/API/preload.php instead I believe

File: inc/main.php [hunk 2]
```diff
@@ -44,7 +44,7 @@ function rocket_init() {
 	require WP_ROCKET_FUNCTIONS_PATH . 'files.php';
 	require WP_ROCKET_FUNCTIONS_PATH . 'posts.php';
 	require WP_ROCKET_FUNCTIONS_PATH . 'admin.php';
-	require WP_ROCKET_FUNCTIONS_PATH . 'preload.php';
+	require WP_ROCKET_INC_PATH . '/Engine/Preload/preload.php';
```
Comment (Tabrisrp) [inline inc/main.php]:
```suggestion
	require WP_ROCKET_INC_PATH . '/API/preload.php';
```

Comment (hellofromtonya) [reply inc/main.php]: @Tabrisrp Are we using these functions to power parts of Rocket's Engine, i.e. the sourcecode internal to it? Or are they for external themes/plugins/etc to use to trigger things within Rocket?

Comment (hellofromtonya) [reply inc/main.php]: Looks like external stuff. I think you're right. I'll move now.

File: inc/Engine/Preload/preload.php (hunk 1 above)
Comment (hellofromtonya) [reply inc/Engine/Preload/preload.php]: Resolved here https://github.com/wp-media/wp-rocket/pull/2396/commits/8675a6bc19165f5173f4f6d11bbd6e13bcfc005c

File: inc/main.php (hunk 2 above)
Comment (hellofromtonya) [reply inc/main.php]: Resolved here https://github.com/wp-media/wp-rocket/pull/2396/commits/8675a6bc19165f5173f4f6d11bbd6e13bcfc005c

**Comment by hellofromtonya** [comment]
Hey @piotrbak @arunbasillal or @vmanthos, this PR is ready for QA. 

It's built off of 3.5.0.2. No feature changes. No bug fixes. It's re-architecting the Preload files to bundle them together as one module. For QA, please make sure that everything still works as it does in 3.5.0.2. 

Thank you!!

**Comment by vmanthos** [comment]
Everything is working the same way as in the version without the bundle.

I've discovered that when updating the content of a post, partial preloading creates only the desktop version and not both the desktop and mobile version as I'd expect.

To reproduce:
1. Enable "Separate cache for mobile devices".
2. Update a post.
3. Check the post's folder.

**Comment by hellofromtonya** [comment]
>I've discovered that when updating the content of a post, partial preloading creates only the desktop version and not both the desktop and mobile version as I'd expect.

@vmanthos Was this issue introduced in this PR? Or is a Mobile Preload issue? If a Mobile Preload issue and not introduced in this PR, please open a new issue. We'll want to address it separately.

**Comment by hellofromtonya** [comment]
Also @vmanthos if the issue you noted was not introduced in this PR and everything works as expected, please approve this PR. Thanks.

**Comment by vmanthos** [comment]
It's a separate issue related to **Mobile Preload**.
I'm already in the process of creating a new issue about this.

**Comment by hellofromtonya** [comment]
Good job @vmanthos identifying this unexpected behavior! 🥇

**Comment by Screenfeed** [comment]
@hellofromtonya This occurs to me now because of the conversation with Arun: since we are changing the namespaces for this module, shouldn't we take the opportunity to also remove the underscores from file names and classes?

**Comment by hellofromtonya** [comment]
@Screenfeed Well that's a good point. You're right. We should. 

I'm thinking, we might want to keep `WP_Rocket` for now though, but change everything else. What do you think?

**Comment by Screenfeed** [comment]
Yes, sure.

**Comment by hellofromtonya** [comment]
Would you like to handle changing this in a separate PR or as part of your counter PR? @Screenfeed

**Comment by Screenfeed** [comment]
Two PRs mean twice more QA. So I'll bundle it in my counter PR.

**Comment by Screenfeed** [comment]
Also, should we create aliases for the old classes, to link to the new ones?

**Comment by hellofromtonya** [comment]
>Also, should we create aliases for the old classes, to link to the new ones?

Good idea for backwards compatibility. Are you thinking of adding those into deprecated 3.5?

**Comment by Screenfeed** [comment]
It seems to be a good place indeed.

**Pull request merged**